sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -205,7 +205,14 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
    * for visualization.
    */
   protected final def executeQuery[T](query: => T): T = {
-    RDDOperationScope.withScope(sparkContext, nodeName, false, true) {
+    val nodeNameScope =
+      if (nodeName == "WholeStageCodegen") {
```

Comment:
It is not a good idea to use string comparison. Why not using type match similar to [PR-26519](https://github.com/apache/spark/pull/26519/files)?

Reply:
How about making nodeNameScope a property in TreeNode?

Reply:
Indeed match expression is a better way to do this, thanks.
I am not sure about modifying TreeNode, I see this proposed implementation as having a smaller footprint.

Reply:
```scala
    val nodeNameScope = this match {
      case plan: WholeStageCodegenExec => s"${plan.nodeName} (${plan.codegenStageId})"
      case _ => nodeName
    }
```